### ISSUE TITLE
pathOr does not work with optional properties

### ISSUE THREAD

**Issue by epeli** [remeda/remeda#5]
Just found out this lib and it seems sweet!

I tried to playaround with the `pathOr` function but it looks like it does not work if the object properties are optional

```ts
type PartialThing = {
    foo?: {
        bar?: {
            num?: number;
            char?: string;
        };
    };
};

const thing: PartialThing = {};

const val = R.pathOr(thing, ["foo", "bar", "num"], 3);
```

![kuva](https://user-images.githubusercontent.com/225712/44369093-d69ad880-a4dd-11e8-8de3-80d2ab2ab3ad.png)

But I found out that it is possible to workaround it by casting the data to be deeply required:

```ts
type DeepRequired<T> = {
    [P in keyof T]-?: T[P] extends Array<infer U>
        ? Array<DeepRequired<U>>
        : T[P] extends Array<infer U>
            ? Array<DeepRequired<U>>
            : DeepRequired<T[P]>
};


// Type checks!
const val = R.pathOr(
    thing as DeepRequired<typeof thing>,
    ["foo", "bar", "num"],
    3,
);
```

I wonder if it would be possible to bake this into the `pathOr` types? Thanks.

**Comment by BetterCallSky** [comment]
Can you provide your `tsconfig.json` and typescript version?
Do you use `"strictNullChecks": true`?

I copied your code to unit tests, and it worked with `strictNullChecks: false`.

![image](https://user-images.githubusercontent.com/4478684/44399605-81e07780-a548-11e8-92fe-4bebb0f8f967.png)

I will try to make it compatible with strictNullChecks.
Thanks!

**Comment by epeli** [comment]
> Do you use "strictNullChecks": true?

Yes. I actually never use TypeScript without it so I actually forgot it can be turned off  😄 

Here's full config I used


```json5
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "es6",
        "lib": ["es6", "dom"],
        "sourceMap": true,
        "noEmit": true,
        "jsx": "react",
        "moduleResolution": "node",
        "forceConsistentCasingInFileNames": true,
        "strict": true, // enables strictNullChecks and other strict rules
        "esModuleInterop": true,
        "noImplicitReturns": true,
        "noImplicitThis": true,
        "noImplicitAny": true,
        "noUnusedLocals": true
    }
}
```

**Comment by BetterCallSky** [comment]
fixed in 0.0.10
